spec/rubocop/cop/rspec/repeated_example_group_body_spec.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::RSpec::RepeatedExampleGroupBody do
+  subject(:cop) { described_class.new }
+
+  it 'registers an offense for repeated describe body' do
+    expect_offense(<<-RUBY)
+      describe 'doing x' do
+      ^^^^^^^^^^^^^^^^^^^^^ Repeated describe block body on line(s) [5]
+        it { cool_predicate_method }
+      end
+
+      describe 'doing y' do
+      ^^^^^^^^^^^^^^^^^^^^^ Repeated describe block body on line(s) [1]
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'registers an offense for repeated context body' do
+    expect_offense(<<-RUBY)
+      context 'when awesome case' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block body on line(s) [5]
+        it { cool_predicate_method }
+      end
+
+      context 'when another awesome case' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block body on line(s) [1]
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'does not register offense for different block body implementation' do
+    expect_no_offenses(<<-RUBY)
+      context 'when awesome case' do
+        it { cool_predicate_method }
+      end
+
+      context 'when another awesome case' do
+        it { another_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'does not register offense if metadata is different' do
+    expect_no_offenses(<<-RUBY)
+      describe 'doing x' do
+        it { cool_predicate_method }
+      end
+
+      describe 'doing x', :request do
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'registers offense for different groups' do
+    expect_offense(<<-RUBY)
+      describe 'doing x' do
+      ^^^^^^^^^^^^^^^^^^^^^ Repeated describe block body on line(s) [5]
+        it { cool_predicate_method }
+      end
+
+      context 'when a is true' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block body on line(s) [1]
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'does not register offense for example groups in different groups' do
+    expect_no_offenses(<<-RUBY)
+      describe 'A' do
+        describe '.b' do
+          context 'when this' do
+            it { do_something }
+          end
+        end
+        context 'when this' do
+          it { do_something }
+        end
+      end
+    RUBY
+  end
+
+  it 'registers offense only for RSPEC namespace example groups' do
+    expect_offense(<<-RUBY)
+      helpers.describe 'doing x' do
+        it { cool_predicate_method }
+      end
+
+      RSpec.describe 'doing x' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated describe block body on line(s) [5]
+        it { cool_predicate_method }
+      end
+
+      context 'when a is true' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block body on line(s) [1]
+        it { cool_predicate_method }
+      end
```

Comment:
The reason this fails is that a group with `helpers.describe` is not detected as `several_blocks?`.
It's possible to use children matching in the pattern, please check some examples [here](https://github.com/rubocop-hq/rubocop-rspec/pull/804/files).
I'm not 100% certain, but it might not support `+`, and this would result in example groups often being the only ones. It's worth to double-check, maybe it does support `+`, I just didn't try.